generator/src/main/java/com/scottlogic/deg/generator/generation/VelocityMonitor.java

```diff
@@ -49,15 +55,42 @@ public void rowEmitted(GeneratedObject row) {
 
     @Override
     public void endGeneration() {
-        reportVelocity(0);
-        System.out.println("\n\nGeneration finished at: " + simpleDateFormat.format(new Date()));
+        timer.cancel();
+
+        LocalDateTime finished = LocalDateTime.now();
+        Duration totalDuration = Duration.between(startedGenerating, finished);
+
+        BigDecimal nanoSecondsAsMilliseconds = BigDecimal
```

Comment:
Would this be clearer by using `getUnits()` and then dealing with the two temporal units together?

Reply:
I think this could confuse the intent/behaviour of this method; its already pretty confusing at points. I'd prefer to leave as-is - but add documentation to clarify the purpose.